lib/pushnotificationsservice.js

```diff
@@ -0,0 +1,350 @@
+var path = require('path');
+var Utils = require('./common/utils');
+var Model = require('./model');
+var log = require('npmlog');
+log.debug = log.verbose;
+
+var PUSHNOTIFICATIONS_TYPES = {
+  'NewCopayer': {
+    filename: 'new_copayer',
+  },
+  'WalletComplete': {
+    filename: 'wallet_complete',
+  },
+  'NewTxProposal': {
+    filename: 'new_tx_proposal',
+  },
+  'NewOutgoingTx': {
+    filename: 'new_outgoing_tx',
+  },
+  'NewIncomingTx': {
+    filename: 'new_incoming_tx',
+  },
+  'TxProposalFinallyRejected': {
+    filename: 'txp_finally_rejected',
+  }
+};
+
+function PushNotificationsService() {};
+
+PushNotificationsService.prototype.start = function(opts, cb) {
+
+  opts = opts || {};
+
+  function _readDirectories(basePath, cb) {
+    fs.readdir(basePath, function(err, files) {
+      if (err) return cb(err);
+      async.filter(files, function(file, next) {
+        fs.stat(path.join(basePath, file), function(err, stats) {
+          return next(!err && stats.isDirectory());
+        });
+      }, function(dirs) {
+        return cb(null, dirs);
+      });
+    });
+  };
+
+  var self = this;
+  self.templatePath = path.normalize(((__dirname + '/templates')) + '/');
+  self.defaultLanguage = 'en';
+  self.defaultUnit = 'btc';
+  self.subjectPrefix = '';
+  self.publicTxUrlTemplate = {};
+
+  async.parallel([
+    function(done) {
+      _readDirectories(self.templatePath, function(err, res) {
+        self.availableLanguages = res;
+        done(err);
+      });
+    },
+    function(done) {
+      self.storage = new Storage();
+      self.storage.connect(opts.storageOpts, done);
+    },
+    function(done) {
+      self.messageBroker = opts.messageBroker || new MessageBroker(opts.messageBrokerOpts);
+      self.messageBroker.onMessage(_.bind(self._sendPushNotifications, self));
+      done();
+    },
+  ], function(err) {
+    if (err) {
+      log.error(err);
+    }
+    return cb(err);
+  });
+};
+
+PushNotificationsService.prototype._sendPushNotifications = function(notification, cb) {
+  var self = this;
+  var url = 'http://192.168.1.143:8000/send';
+  cb = cb || function() {};
+
+  var notifType = PUSHNOTIFICATIONS_TYPES[notification.type];
+  if (!notifType) return cb();
+
+  self._getRecipientsList(notification.walletId, function(err, recipientsList) {
+    self.storage.fetchWallet(notification.walletId, function(err, wallet) {
+
+      var resultedRecipientsList = _.reject(self._getJoinedRecipientsList(wallet, recipientsList), {
+        id: notification.creatorId || null
+      });
+      async.waterfall([
+        function(next) {
+          self._readAndApplyTemplates(notification, notifType, resultedRecipientsList, next);
+        },
+        function(contents, next) {
+          async.map(resultedRecipientsList, function(recipient, next) {
+            var opts = {};
+            var content = contents[recipient.language];
+            opts.users = [notification.walletId + '$' + recipient.id];
```

Comment:
Seeing that the `users` field can receive an array of users, a nice optimization (to reduce the number of POSTS) would be to group users by language and send the same notification to all users under each language.